feature/feature.go

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright 2023 SUSE.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package feature
+
+import (
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/component-base/featuregate"
+)
+
+const (
+	// RancherKubeSecretPatch is used to enable patching of the Rancher v2prov created kubeconfig
+	// secrets so that they can be used with CAPI 1.5.x.
+	RancherKubeSecretPatch featuregate.Feature = "rancher-kube-secret-patch" //nolint:gosec
+)
+
+func init() {
+	utilruntime.Must(MutableGates.Add(defaultGates))
+}
+
+var defaultGates = map[featuregate.Feature]featuregate.FeatureSpec{
+	RancherKubeSecretPatch: {Default: false, PreRelease: featuregate.Beta},
```

Comment:
```suggestion
	RancherKubeSecretPatch: {Default: true, PreRelease: featuregate.Beta},
```

Reply:
My reasoning for disabling it here was:
- if someone uses the controller themeselves then the feature will be turned off
- we explicitly enable it in the values.yaml, with the idea in the future we can just toggle it to false in the chart

Reply:
no feature gate should be enabled by default, this is why we have them.

Reply:
@alexander-demicev Read the https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/#feature-stages Features in beta stage are “usually” enabled by default. Feature gating goal, is to graduate a feature. Not eliminate one with the passage of time, which is a plan for this implementation. And JIC, the PR was already approved, so this discussion is irrelevant.